ntp.go

```diff
@@ -218,9 +218,9 @@ func (r *Response) RootDistance() time.Duration {
 }
 
 // CausalityViolation returns a time duration representing the amount of
-// causality violation between SNTP timestamps. It may be used as a lower bound
-// on the local clock time synchronization error.  NB: The "legit" error may be
-// up to 1s during leap second.
+// causality violation between two sets of timestamps. It may be used as a
+// lower bound on current time synchronization error betwen local and NTP
+// clock. A leap second may contribute as much as 1 second of causality violation.
 func (r *Response) CausalityViolation() time.Duration {
```

Comment:
Did you disagree with my request to make `causalityViolation` and `rootDistance` fields of `Response` instead of methods?

Also, I just noticed that you're using `err` to hold the "violation error" value.  Could you rename this variable?  Idiomatic code uses the `err` variable to hold a go error response.

Reply:
> Did you disagree with my request to make `causalityViolation` .. fields .. instead of methods

I think that _data structure invariant consistency_ is a good argument to keep them as methods. Don't you think so?

I don't insist. If you consider that argument weak, I'm OK converting them to fields. I just want to say that I've not thought of that argument when I said _"I have no strong opinion if RTT/Offset or T1..4 should be stored"_.

> you're using `err` to hold the ...

Yep, that's obviously wrong name, thanks for catching.

Reply:
I mean, please, state that explicitly if you still insist on converting methods to fields :-)

Reply:
>I think that data structure invariant consistency is a good argument to keep them as methods. Don't you think so?

I understand your rationale now.  

I guess I'm not that concerned programs changing the fields of the `Response` structure, because it's a result returned from a function, and I don't expect anyone would have a reason to modify it after it's returned.

>I mean, please, state that explicitly if you still insist on converting methods to fields :-)

Yes, I'd still like them converted to fields instead of methods.  Apologies if this creates a bunch of work for you.  I also realize this change will make it harder to document the new fields in a way that is similarly concise as the other fields' documentation, but please feel free to switch to a multi-line prefacing comment for each Response field.

Reply:
Another side-note. I've just noticed, that I've probably followed the style of `time` unconsciously. `time` stores time in quite opaque structure and `Seconds()`, `Unix()` and `UnixNano()` are callable converters.

Anyway, these 8..16 bytes of RAM do not matter that much to discuss their destiny that long :-)